datafusion/expr/src/utils.rs

```diff
@@ -909,8 +909,8 @@ pub fn check_all_columns_from_schema(
 pub fn find_valid_equijoin_key_pair(
     left_key: &Expr,
     right_key: &Expr,
-    left_schema: DFSchemaRef,
-    right_schema: DFSchemaRef,
+    left_schema: &DFSchema,
```

Comment:
thats interesting, should we a reference instead of Arc whenever possible?

Reply:
I think in general we should use a reference when the underlying callsite doesn't actually need a owned copy

So in this case, the callsite just needs a `&DFSchema` so there is no need to go through the overhead of creating the `Arc`

I think the runtime overhead of creating another Arc is likely to be unmeasurable in all but the most extreme circumstances. However I think the code is often easier to reason about